Makefile

```diff
@@ -1,252 +1,62 @@
-# the dependencies listed in the "package.json" file.
-node_modules: package.json | node-version
-	@$(NPM) prune
-	@$(NPM) install
-	@touch node_modules
-
-test: build
-	@$(NPM) test
+install:
+	npm run install-if-needed
 
-lint: node_modules/eslint node_modules/eslint-plugin-react node_modules/babel-eslint config-defaults-lint mixedindentlint
-	@$(NPM) run lint
+run:
+	npm start
 
-eslint: lint
+dashboard:
+	@npm run dashboard
 
-eslint-branch: node_modules/eslint node_modules/eslint-plugin-react node_modules/babel-eslint
-	@git diff --name-only $$(git merge-base $$(git rev-parse --abbrev-ref HEAD) master)..HEAD | grep '\.jsx\?$$' | xargs $(NODE_BIN)/eslint --cache
+test:
+	@npm test
 
-# Skip files that are auto-generated
-mixedindentlint: node_modules/mixedindentlint
-	@echo "$(JS_FILES)\n$(SASS_FILES)" | xargs $(NODE_BIN)/mixedindentlint --ignore-comments --exclude="client/config/index.js"
+lint:
+	@npm run lint
 
-# Ensure that default config values exist in _shared.json
-config-defaults-lint: $(CONFIG_FILES)
-	@$(NODE) bin/validate-config-keys.js || exit
+eslint:
+	@npm run lint
 
-# keep track of the current CALYPSO_ENV so that it can be used as a
-# prerequisite for other rules
-.env: FORCE
-	@$(RECORD_ENV) $@
+eslint-branch:
+	@npm run eslint-branch
 
-public/style.css: node_modules $(SASS_FILES)
-	@$(SASS) assets/stylesheets/style.scss $@
-	@$(AUTOPREFIXER) $@
+mixedindentlint:
+	@npm run lint:mixedindent
 
-public/style-debug.css: node_modules $(SASS_FILES)
-	@$(SASS) --source-map "$(@D)/style-debug.css.map" assets/stylesheets/style.scss $@
-	@$(AUTOPREFIXER) $@
+config-defaults-lint:
+	@npm run lint:config-defaults
 
-public/style-rtl.css: public/style.css
-	@$(RTLCSS) $(THIS_DIR)/public/style.css $@
+build-server:
+	@npm run build-server
 
-public/editor.css: node_modules $(SASS_FILES)
-	@$(SASS) assets/stylesheets/editor.scss $@
-	@$(AUTOPREFIXER) $@
+build:
+	@npm run build
 
-public/directly.css: node_modules $(SASS_FILES)
-	@$(SASS) assets/stylesheets/directly.scss $@
-	@$(AUTOPREFIXER) $@
+build-desktop:
+	@npm run build-desktop
 
-server/devdocs/search-index.js: $(MD_FILES) $(ALL_DEVDOCS_JS)
-	@$(ALL_DEVDOCS_JS) $(MD_FILES)
-
-server/devdocs/components-usage-stats.json: $(COMPONENTS_USAGE_STATS_FILES) $(COMPONENTS_USAGE_STATS_JS)
-	@$(COMPONENTS_USAGE_STATS_JS) $(COMPONENTS_USAGE_STATS_FILES)
-
-server/devdocs/proptypes-index.json: $(COMPONENTS_PROPTYPE_FILES) $(COMPONENTS_PROPTYPES_JS)
-	@$(COMPONENTS_PROPTYPES_JS) $(COMPONENTS_PROPTYPE_FILES)
-
-build-server: install
-	@mkdir -p build
-	@CALYPSO_ENV=$(CALYPSO_ENV) $(NODE_BIN)/webpack --display-error-details --config webpack.config.node.js
+clean:
+	@npm run clean
 
-build: install build-server build-css server/devdocs/search-index.js server/devdocs/proptypes-index.json server/devdocs/components-usage-stats.json
-	@if [ $(CALYPSO_ENV) != development ]; then $(BUNDLER); fi
+distclean:
+	@npm run distclean
```

Comment:
🤦‍♂️ I've never tried to run `clean` with an already empty `node_modules`. You're right, those commands shouldn't need any packages to run. Good catch!

Fixed in https://github.com/Automattic/wp-calypso/pull/14631/commits/76a2b260bcb47e0cfd630f294218c48fb723bf68